internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -294,6 +294,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		"external-name", meta.GetExternalName(cm),
 	)
 
+	res, err := r.reconcileClaim(ctx, log, record, cm)
+	if err != nil {
+		cm.SetConditions(xpv1.ReconcileError(err))
+	} else {
+		cm.SetConditions(xpv1.ReconcileSuccess())
+	}
+
+	return res, errors.Wrap(r.client.Status().Update(ctx, cm), errUpdateClaimStatus)
+}
```

Comment:
I'd prefer to update the reconciler to set `ReconcileError` as part of the existing `SetConditions` calls, rather than moving the bulk of the logic into a wrapped `reconcile` method. The former approach is more in line with how we approach this in the managed reconciler, for example:

https://github.com/crossplane/crossplane-runtime/blob/988c9ba9c255d7cade32e2c0135ce47b083eaf95/pkg/reconciler/managed/reconciler.go#L684